xdl_rs/src/conversions/base_type.rs

```diff
@@ -11,18 +11,39 @@ impl From<XdlPrimitiveMetadata> for XdlMetadata {
     }
 }
 
+// TODO: Test this
```

Comment:
Do you you want me to merge this before test cases are in? Not sure how hard it would be to test this.

Reply:
If you want me to merge now and then you can do this later, all good. Just wanted to get confirmation before I hit the button.

Reply:
Yeah i think that should work, as long as you're validating state after (and/or before?). Actually using the stuff is a good way to do unit testing i think, since that can catch weird edge cases that the monotonous ones don't.

Reply:
Added tests for the stuff we talked about (except for struct, as its not done yet). I also did some wacky trait things that I think you'll enjoy :)